MlDistributedFailureIT#testClusterWithTwoMlNodes_RunsDatafeed_GivenOriginalNodeGoesDown failures

<!--
Please fill out the following information, and ensure you have attempted
to reproduce locally
-->

**Build scan**:
https://gradle-enterprise.elastic.co/s/3eakwbhmebz7q
**Repro line**:
```
./gradlew ':x-pack:plugin:ml:internalClusterTest' --tests "org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.testClusterWithTwoMlNodes_RunsDatafeed_GivenOriginalNodeGoesDown" -Dtests.seed=F8D217ED9786DD6 -Dtests.security.manager=true -Dtests.locale=es-MX -Dtests.timezone=Etc/GMT-1 -Druntime.java=11
```
**Reproduces locally?**:
Didn't reproduce locally on master

**Applicable branches**:
master, 7.x, 7.11

**Failure history**:
https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(branch),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:testClusterWithTwoMlNodes_RunsDatafeed_GivenOriginalNodeGoesDown),sort:!(time,desc))
**Failure excerpt**:
```
org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT > testClusterWithTwoMlNodes_RunsDatafeed_GivenOriginalNodeGoesDown FAILED
    java.lang.AssertionError: expected:<opened> but was:<closed>
        at __randomizedtesting.SeedInfo.seed([F8D217ED9786DD6:A8C0156C00DDD93D]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$waitForJobClosed$21(MlDistributedFailureIT.java:657)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:955)
        at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.waitForJobClosed(MlDistributedFailureIT.java:655)
        at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.testClusterWithTwoMlNodes_RunsDatafeed_GivenOriginalNodeGoesDown(MlDistributedFailureIT.java:510)
REPRODUCE WITH: ./gradlew ':x-pack:plugin:ml:internalClusterTest' --tests "org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.testClusterWithTwoMlNodes_RunsDatafeed_GivenOriginalNodeGoesDown" -Dtests.seed=F8D217ED9786DD6 -Dtests.security.manager=true -Dtests.locale=es-MX -Dtests.timezone=Etc/GMT-1 -Druntime.java=11
```


Going to mute this

--- a/x-pack/plugin/ml/src/internalClusterTest/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java
+++ b/x-pack/plugin/ml/src/internalClusterTest/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java
@@ -85,7 +85,9 @@ import static org.hamcrest.Matchers.arrayWithSize;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
 import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
 
 public class MlDistributedFailureIT extends BaseMlIntegTestCase {
 
@@ -467,7 +469,6 @@ public class MlDistributedFailureIT extends BaseMlIntegTestCase {
         });
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/67756")
     public void testClusterWithTwoMlNodes_RunsDatafeed_GivenOriginalNodeGoesDown() throws Exception {
         internalCluster().ensureAtMostNumDataNodes(0);
         logger.info("Starting dedicated master node...");
@@ -513,18 +514,24 @@ public class MlDistributedFailureIT extends BaseMlIntegTestCase {
         setMlIndicesDelayedNodeLeftTimeoutToZero();
 
         StartDatafeedAction.Request startDatafeedRequest = new StartDatafeedAction.Request(config.getId(), 0L);
-        startDatafeedRequest.getParams().setEndTime("now");
         client().execute(StartDatafeedAction.INSTANCE, startDatafeedRequest).get();
 
         waitForJobToHaveProcessedAtLeast(jobId, 1000);
 
         internalCluster().stopNode(nodeRunningJob.getName());
 
-        waitForJobClosed(jobId);
+        // Wait for job and datafeed to get reassigned
+        assertBusy(() -> {
+            assertThat(getJobStats(jobId).getNode(), is(not(nullValue())));
+            assertThat(getDatafeedStats(datafeedId).getNode(), is(not(nullValue())));
+        }, 30, TimeUnit.SECONDS);
 
-        DataCounts dataCounts = getJobStats(jobId).getDataCounts();
-        assertThat(dataCounts.getProcessedRecordCount(), greaterThanOrEqualTo(numDocs));
-        assertThat(dataCounts.getOutOfOrderTimeStampCount(), equalTo(0L));
+        assertBusy(() -> {
+            DataCounts dataCounts = getJobStats(jobId).getDataCounts();
+            assertThat(dataCounts.getProcessedRecordCount(), greaterThanOrEqualTo(numDocs));
+            assertThat(dataCounts.getOutOfOrderTimeStampCount(), equalTo(0L));
+        });
+
     }
 
     private void setupJobWithoutDatafeed(String jobId, ByteSizeValue modelMemoryLimit) throws Exception {

